src/v/redpanda/admin_server.cc

```diff
@@ -1058,6 +1056,41 @@ void config_multi_property_validation(
                   = "Must be set when cloud storage enabled";
             }
         }
+
+        if (
```

Comment:
Can we simplify this to just construct `properties` with a different set of required fields depending on the auth mode, rather than writing out the `if` blocks?

It makes the errors a little bit less descriptive, but should be more robust to maintain.  We can change the "Must be set when cloud storage enabled" to just "Must be set" and I think it's a sufficient feedback to the user that they missed a field.

Reply:
Makes sense, I tried to make it a bit more generic.